services-geojson/src/commonMain/kotlin/org/maplibre/geojson/serializer/PointDoubleArraySerializer.kt

```diff
@@ -0,0 +1,35 @@
+package org.maplibre.geojson.serializer
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.builtins.DoubleArraySerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import org.maplibre.geojson.model.Point
+
+/**
+ * Internal serializer/deserializer that is converting a [Point] object into a
+ * double array (aka GeoJSON Position).
+ */
+internal class PointDoubleArraySerializer : KSerializer<Point> {
+    private val delegateSerializer = DoubleArraySerializer()
+
+    @OptIn(ExperimentalSerializationApi::class)
+    override val descriptor = SerialDescriptor("Point", delegateSerializer.descriptor)
+
+    override fun serialize(encoder: Encoder, value: Point) {
+        val data = listOfNotNull(
+            value.longitude,
+            value.latitude,
+            value.altitude
+        )
+
+        encoder.encodeSerializableValue(delegateSerializer, data.toDoubleArray())
+    }
+
+    override fun deserialize(decoder: Decoder): Point {
+        val array = decoder.decodeSerializableValue(delegateSerializer)
+        return Point(longitude = array[0], latitude = array[1], altitude = array.getOrNull(2))
+    }
```

Comment:
If you anyway always construct `Point`s from JSON with this constructor, then wouldn't it make sense to have `longitude`, `latitude` and `altitude` as fields in `Point` instead of this (oddly named) `coordinates: List<Double>`?

See also https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.1

Reply:
A point is a GeoJSON object like this:

```json
{
  "type": "Point",
  "coordinates": [100.0, 0.0]
}
```

or is used as inner object only as array. That is the reason, that this serializer is exists.

```json
{
  "type": "LineString",
  "coordinates": [
     [100.0, 0.0], // this is also a point in our models
     [101.0, 1.0] // this is also a point in our models
  ]
}
```


But yeah, I will overthink the model structure based on your suggestions.